src/vm/dllimport.cpp

```diff
@@ -6438,6 +6474,51 @@ NATIVE_LIBRARY_HANDLE NDirect::LoadLibraryModuleViaEvent(NDirectMethodDesc * pMD
     return hmod;
 }
 
+NATIVE_LIBRARY_HANDLE NDirect::LoadLibraryModuleViaCallback(NDirectMethodDesc * pMD, LPCWSTR wszLibName)
+{
+    STANDARD_VM_CONTRACT;
+
+    if (pMD->GetModule()->IsSystem())
+    {
+        // Don't attempt to callback on Corelib itself.
+        // The LoadLibrary callback stub is managed code that requires CoreLib 
+        return NULL;
+    }
+
+    DWORD dllImportSearchPathFlags;
+    BOOL searchAssemblyDirectory;
+    BOOL hasDllImportSearchPathFlags = GetDllImportSearchPathFlags(pMD, &dllImportSearchPathFlags, &searchAssemblyDirectory);
```

Comment:
If this is false, then `dllImportSearchPathFlags` and `searchAssemblyDirectory` are in an undefined state. Please handle this case.

Reply:
If `GetDllImportSearchPathFlags` returns false, `dllImportSearchPathFlags` and `searchAssemblyDirectory` are initialized to default values.

But the contract with managed code is that `dllImportSearchPathFlags` is meaningless if `hasDllImportSearchPathFlags` is false. `LoadLibraryCallbackStub` will pass `null` to the resolver.

Reply:
Yeah I am not sure what was happening here. I must have misread the code. Ignore.